ui/packages/catalog-ui-search/src/main/webapp/component/sharing/sharing.view.js

```diff
@@ -0,0 +1,328 @@
+    .map(email => {
+      return {
+        id: common.generateUUID(),
+        type: 'user',
+        access: accessAdministrators.includes(email)
+          ? 'share'
+          : individuals.includes(email)
+            ? 'write'
+            : individualsRead.includes(email)
+              ? 'read'
+              : 'none',
+        value: email,
+      }
+    })
+}
+
+export default class WorkspaceSharing extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      items: [],
+    }
+  }
+
+  componentDidMount() {
+    fetch(`/search/catalog/internal/metacard/${this.props.id}`)
+      .then(response => response.json())
+      .then(data => {
+        const metacard = data.metacards[0]
+        const items = getGroups(
+          metacard['security.access-groups'],
+          metacard['security.access-groups-read']
+        ).concat(
+          getIndividuals(
+            metacard['metacard.owner'],
+            metacard['security.access-individuals'],
+            metacard['security.access-individuals-read'],
+            metacard['security.access-administrators']
+          )
+        )
+        this.setState({ items: items })
+        this.add()
+      })
+  }
+
+  save() {
+    const roles = this.state.items.filter(e => e.type === 'role')
+    const users = this.state.items.filter(
+      e => e.value !== '' && e.type === 'user'
+    )
+
+    fetch(`/search/catalog/internal/metacards`, {
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify([
+        {
+          ids: [this.props.id],
+          attributes: [
+            {
+              attribute: 'security.access-individuals',
+              values: users.filter(e => e.access === 'write').map(e => e.value),
+            },
+            {
+              attribute: 'security.access-individuals-read',
+              values: users.filter(e => e.access === 'read').map(e => e.value),
+            },
+            {
+              attribute: 'security.access-groups',
+              values: roles.filter(e => e.access === 'write').map(e => e.value),
+            },
+            {
+              attribute: 'security.access-groups-read',
+              values: roles.filter(e => e.access === 'read').map(e => e.value),
+            },
+            {
+              attribute: 'security.access-administrators',
+              values: users.filter(e => e.access === 'share').map(e => e.value),
+            },
+          ],
+        },
+      ]),
+    })
+      .then(res => res.json())
+      .then(() => {
+        this.props.lightbox.close()
+        announcement.announce(
+          {
+            title: 'Success!',
+            message: 'Sharing saved',
+            type: 'success',
+          },
+          1500
+        )
+      })
+      .catch(function() {
+        announcement.announce(
+          {
+            title: 'Error',
+            message: 'Save failed',
```

Comment:
This could be more descriptive. What about `Could not update the workspace access controls`?